packages/data-broker/src/DataBroker.js

```diff
@@ -34,6 +46,21 @@ export class DataBroker {
       [this.getDataSourceTypes().DATA_GROUP]: this.fetchFromDataGroupTable,
       [this.getDataSourceTypes().SYNC_GROUP]: this.fetchFromSyncGroupTable,
     };
+    this.permissionCheckers = {
```

Comment:
It's not entirely obvious why data broker is doing permission checks. 

You could think that it should be done before or after the data broker. 
- Before: different callers may use data broker differently. In some cases it's for fetching data to show to a user, in other cases it may be a machine user which needs access to all data.
- After: DHIS already has permission checks, why do it twice?

I'm guessing the reasoning is that we want the services to expose all data, and for data broker to be user-aware and only expose the data the user is allowed to see. If so, this would be really good to comment here somewhere for the next devs coming across this.

Reply:
I think it makes sense if you think about it more. Happy with the small comment.